src/time.cr

```diff
@@ -1359,9 +1359,20 @@ struct Time
 
   # Returns a copy of this `Time` representing the beginning of the week.
   #
+  # The week starts on Monday by default, but can be configured by passing a different `start_day` as a `Time::DayOfWeek`.
+  #
+  # ```
+  # now = Time.local                                  # => 2023-05-08 16:48:13 +00:00
+  # now.at_beginning_of_week(Time::DayOfWeek::Monday) # => 2023-05-08 00:00:00 +00:00
+  # now.at_beginning_of_week(Time::DayOfWeek::Sunday) # => 2023-05-07 00:00:00 +00:00
+  #
+  # # with a symbol
+  # now.at_beginning_of_week(:monday) # => 2023-05-08 00:00:00 +00:00
+  # now.at_beginning_of_week(:sunday) # => 2023-05-07 00:00:00 +00:00
+  # ```
   # TODO: Ensure correctness in local time-line.
-  def at_beginning_of_week : Time
-    (self - (day_of_week.value - 1).days).at_beginning_of_day
+  def at_beginning_of_week(start_day : Time::DayOfWeek = Time::DayOfWeek::Monday) : Time
```

Comment:
You can use auto-casting already at the definition.

```suggestion
  def at_beginning_of_week(start_day : Time::DayOfWeek = :monday) : Time
```

Reply:
Thanks!